### PR TITLE
feat(web-components): migrate from npmignore to package.json#files for release

### PR DESCRIPTION
**Pull request by Hotell**
## New Behavior

- migrates from `.npmignore` to `package.json#files` in order to declare what will be shipped to npm
- adds integration test via `verify-packaging` task target

**Before**

image

**After**

image

## Related Issue(s)

- Fixes partially https://github.com/microsoft/fluentui/issues/31349

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: apps/vr-tests-web-components/project.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "name": "@fluentui/vr-tests-web-components",
```
Comment (fabricteam) [inline apps/vr-tests-web-components/project.json:2]: ## [🕵 fluentui-web-components-v3 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31519&commitId=b7c89db01335f142c6be07ce05ceb78ccb0d00e0&env=prod&clientType=FLUENTUI)


  <details><summary><b>Accordion</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Accordion.Size - RTL.normal.chromium.png|2| Changed|
|Accordion.Size - Dark Mode.normal.chromium.png|2| Changed|
|Accordion.Size.normal.chromium.png|2| Changed|

</details>


  <details><summary><b>Avatar</b> 5 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar.Active Appearance.normal.chromium.png|10| Changed|
|Avatar.Color - RTL.normal.chromium.png|1| Changed|
|Avatar.Active.normal.chromium.png|5| Changed|
|Avatar.Active Appearance - Dark Mode.normal.chromium.png|2| Changed|
|Avatar.Color.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Badge</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Badge.Shape.normal.chromium.png|1| Changed|
|Badge.Default.normal.chromium.png|1| Changed|
|Badge.Appearance - Dark Mode.normal.chromium.png|15| Changed|
|Badge.Appearance.normal.chromium.png|1| Changed|
|Badge.Size.normal.chromium.png|6| Changed|
|Badge.Color.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Button</b> 15 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button.Large Icon Only - Dark Mode.default.chromium.png|4| Changed|
|Button.Large Icon Only.pressed.chromium.png|3| Changed|
|Button.Large Icon Only - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large Icon Only.hover.chromium.png|4| Changed|
|Button.Large With Icon - Dark Mode.hover.chromium.png|3| Changed|
|Button.Large With Icon - Dark Mode.default.chromium.png|4| Changed|
|Button.Large With Icon - Dark Mode.pressed.chromium.png|5| Changed|
|Button.Large With Icon.hover.chromium.png|4| Changed|
|Button.Large With Icon.pressed.chromium.png|3| Changed|
|Button.Large With Icon.default.chromium.png|5| Changed|
|Button.Large With Icon - RTL.default.chromium.png|5| Changed|
|Button.Large With Icon - RTL.pressed.chromium.png|3| Changed|
|Button.Large Icon Only.default.chromium.png|5| Changed|
|Button.Large With Icon - RTL.hover.chromium.png|4| Changed|
|Button.Large Icon Only - Dark Mode.hover.chromium.png|3| Changed|

</details>


  <details><summary><b>Divider</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Divider.With Svg.normal.chromium.png|3| Changed|
|Divider.With Svg - RTL.normal.chromium.png|3| Changed|
|Divider.Vertical With Svg.normal.chromium.png|4| Changed|

</details>


  <details><summary><b>MenuList</b> 16 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|MenuList.Checkbox.2nd selected.chromium.png|1| Changed|
|MenuList.Checkbox - RTL.2nd selected.chromium.png|1| Changed|
|MenuList.Checkbox With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.1st selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.normal.chromium.png|4| Changed|
|MenuList.Checkbox With Icons - RTL.1st selected.chromium.png|4| Changed|
|MenuList.Checkbox With Icons.2nd selected.chromium.png|5| Changed|
|MenuList.Radio With Icons.1st selected.chromium.png|4| Changed|
|MenuList.Radio With Icons.normal.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.normal.chromium.png|4| Changed|
|MenuList.Checkbox With Icons - RTL.2nd selected.chromium.png|5| Changed|
|MenuList.Radio With Icons.2nd selected.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.1st selected.chromium.png|4| Changed|
|MenuList.Radio With Icons - RTL.2nd selected.chromium.png|4| Changed|
|MenuList.With Icons.hover menuitem.chromium.png|3| Changed|
|MenuList.With Icons.normal.chromium.png|4| Changed|

</details>


  <details><summary><b>Slider</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Slider.Size Small.rightArrow.chromium.png|1| Changed|
|Slider.Size Small.normal.chromium.png|1| Changed|

</details>


  <details><summary><b>Switch</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Switch.Checked - RTL.hover.chromium.png|1| Changed|
|Switch.Checked.hover.chromium.png|1| Changed|

</details>


  <details><summary><b>Text</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Text.Block.normal.chromium.png|995| Changed|

</details>


  <details><summary><b>TextInput</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|TextInput.Size Large.normal.chromium.png|2| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: packages/web-components/project.json [hunk 2]
```diff
@@ -2,5 +2,6 @@
   "name": "@fluentui/web-components",
   "$schema": "../../node_modules/nx/schemas/project-schema.json",
   "projectType": "library",
-  "implicitDependencies": []
+  "implicitDependencies": [],
+  "tags": ["platform:web", "web-components"]
```
Comment (Hotell) [inline packages/web-components/project.json:6]: following our package patterns to define domain

File: packages/web-components/scripts/type-check.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
 // @ts-check
 
 import fs from 'node:fs';
-import { fileURLToPath } from 'node:url';
 import path from 'node:path';
 import { promisify } from 'node:util';
 import { exec } from 'node:child_process';
 import { exit } from 'node:process';
 
+import { getDirname } from './utils.js';
```
Comment (Hotell) [inline packages/web-components/scripts/type-check.js:9]: extracted this util to shared module as this pattern is used already in more than 1 place

File: packages/web-components/scripts/verify-packaging.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+// @ts-check
+/**
+ * Copied from ${@link 'file://./../../../scripts/tasks/src/verify-packaging.ts'}
```
Comment (Hotell) [inline packages/web-components/scripts/verify-packaging.js:3]: following the pattern - just-scripts doesn't support ESM, we will migrate away from just-scripts anyways so copy-paste for now ( with tweaks )

File: packages/web-components/package.json [hunk 5]
```diff
@@ -18,6 +18,13 @@
   "main": "dist/esm/index.js",
   "types": "dist/web-components.d.ts",
   "unpkg": "dist/web-components.min.js",
+  "files": [
+    "*.md",
```
Comment (radium-v) [inline packages/web-components/package.json:22]: The `README` and `CHANGELOG` (along with any `LICENSE` or `NOTICE` files, and `package.json`) should automatically get included by default, so this entry may not be necessary.

File: packages/web-components/scripts/type-check.js (hunk 3 above)
Comment (radium-v) [reply packages/web-components/scripts/type-check.js:9]: `__dirname` and `__filename` are available in Node >= 20.11.0 as `import.meta.dirname` and `import.meta.filename`

**Review by radium-v — APPROVED** [review]

File: packages/web-components/scripts/type-check.js (hunk 3 above)
Comment (Hotell) [reply packages/web-components/scripts/type-check.js:9]: nice TIL, i'll update , ty!

**Review by janechu — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]

**Review by tudorpopams — APPROVED** [review]

**Pull request merged**